cli/src/commands/branch.rs

```diff
@@ -569,6 +569,55 @@ fn cmd_branch_track(
             names.len()
         )?;
     }
+
+    //show conflicted branches if there are some
+    let template = {
+        let language = workspace_command.commit_template_language()?;
+        let text = command
+            .settings()
+            .config()
+            .get::<String>("templates.branch_list")?;
+        workspace_command
+            .parse_template(&language, &text, CommitTemplateLanguage::wrap_ref_name)?
+            .labeled("branch_list")
+    };
+
+    ui.request_pager();
+    if let Some(mut formatter) = ui.status_formatter() {
+        let branches_to_list =
+            workspace_command
+                .repo()
+                .view()
+                .branches()
+                .filter(|(name, target)| {
+                    names
+                        .iter()
+                        .map(|n| n.branch.clone())
+                        .contains(&name.to_string())
```

Comment:
nit: can be expressed as `.any(|n| n.branch == *name)`.

fwiw, it's probably simpler to build a `HashMap` of `local: {remotes}` first. We'll need to filter remotes by `[name].remotes`, not by `remotes_of_arbitrary_locals`.

Reply:
so this hashmap would contain local branches names as keys and a list of remotes as values ?  isn't it what names look like (except it's a vector but essentially each element has a name and a list of remotes)

Reply:
> so this hashmap would contain local branches names as keys and a list of remotes as values ? isn't it what names look like (except it's a vector but essentially each element has a name and a list of remotes)

Yes. We'll need per-branch remote names to fix filtering of remote refs:
```rust
        for (name, branch_target) in branches_to_list {
            ...
            for (remote_name, remote_ref) in branch_target.remote_refs {
                if names.iter().filter(|x| x.remote == *remote_name).count() > 0 {
                                      // && x.name == name
```

Maybe we can instead rely on the order of `names` to do sorted merge join, (or do linear search), but using `HashMap` would be simpler and easier to follow.

Reply:
sorry but i don't get what we are trying to fix 🤔 

> Yes. We'll need per-branch remote names to fix filtering of remote refs:

names array contains for each branch to track the name of the associated remote. Don't we have per-branch remote name here ?

Reply:
is it better like this ?

Reply:
Well, `names` is basically a list of `(branch_name, remote_name)` pairs, and it can be e.g. `[("main", "origin"), ("main", "upstream")]`, right? So `HashMap<str, str>` wouldn't work.

Reply:
haha ok, should be good now